### ISSUE TITLE
EAT and EAT_AT_INN looks for plates but never places them

### ISSUE THREAD

**Issue by KnightCaptainU7** [exult/exult#270]
The scheduling code for EAT and EAT_AT_INN seems to not place plates, so no one eats. Testing in Monitor, Lucilla makes the "swing" like she's trying to place a plate, but it doesn't land. 

EAT also does not seem to place plates. This is visible in the Rangers' mess hall, where they should place places on the table before placing food on them, but they never do in Exult.

**Comment by KnightCaptainU7** [comment]
In DOSbox, Lucilla places small white places and more often the same size plate but of wood. So there is likely some logic of which plates to use on which tables?

**Comment by KnightCaptainU7** [comment]
The southwest red chair in Sleeping Bull is too far from the table for the NPC seated to be fed. It needs to be over to the east slightly. The north one at the same table also needs to be moved east to center it.

In the Rangers' mess hall, the two chairs below the table are too close to allow any NPC to sit.

**Comment by DominusExult** [comment]
> The southwest red chair in Sleeping Bull is too far from the table for the NPC seated to be fed. It needs to be over to the east slightly. The north one at the same table also needs to be moved east to center it.
> 
> In the Rangers' mess hall, the two chairs below the table are too close to allow any NPC to sit.

this is also not working in the original, right? If so, then I wouldn't worry about it other than SI Fixes. To fix it outside of a mod is very hackish. We have a function for that but it's reserved for critical stuff.

**Comment by KnightCaptainU7** [comment]
Correct, if the chair is too close in the original the NPCs do not move them to sit.

**Comment by KnightCaptainU7** [comment]
Testing in SI in Fawn, the plates still seems to be an issue. Using Exult 1.10.1.

The chairs in some places are issues that can be fixed in the initgame.dat file, so I will do that in my mod.

**Comment by DominusExult** [comment]
@marzojr from "regression testing" with older builds, I found that f4be2947e `Applying clang-tidy's modernize-loop-convert` (or any of the other October 4th 2020) broke placing plates completely. It seems to be brokish before as very few plates are put on the table but it stops completely on this day.

If I place some debug messages in `Game_object* Waiter_schedule::create_customer_plate` (current code) I can observe that the `for (auto& eating_table : eating_tables)` loop is never run.

**Comment by marzojr** [comment]
> return table != nullptr;

This should be `return table == nullptr;`, I think.

**Comment by DominusExult** [comment]
That makes it work better, yes, and in most inn's it works correctly. Just the Blue Boar Britain seems to have more problems.

**Comment by DominusExult** [comment]
Committed your fix but leaving it open because something else is broken as described above.

It *seems* to me that the table shape #964 uncovers some issue we have with computing the distance and where to place plates

**Comment by DominusExult** [comment]
It does* work, my savegame was broken there.
